RNTester/RNTesterUnitTests/RCTConvert_UIColorTests.m

```diff
@@ -0,0 +1,200 @@
+    XCTAssertEqual(rgba1[0], rgba2[0]);
+    XCTAssertEqual(rgba1[1], rgba2[1]);
+    XCTAssertEqual(rgba1[2], rgba2[2]);
+    XCTAssertEqual(rgba1[3], rgba2[3]);
+
+    [UITraitCollection setCurrentTraitCollection:savedTraitCollection];
+  }
+#endif
+}
+
+- (void)testGenerateFallbacks
+{
+  NSDictionary<NSString *, NSNumber*>* semanticColors = @{
+    // https://developer.apple.com/documentation/uikit/uicolor/ui_element_colors
+    // Label Colors
+    @"labelColor": @(0xFF000000),
+    @"secondaryLabelColor": @(0x993c3c43),
+    @"tertiaryLabelColor": @(0x4c3c3c43),
+    @"quaternaryLabelColor": @(0x2d3c3c43),
+    // Fill Colors
+    @"systemFillColor": @(0x33787880),
+    @"secondarySystemFillColor": @(0x28787880),
+    @"tertiarySystemFillColor": @(0x1e767680),
+    @"quaternarySystemFillColor": @(0x14747480),
+    // Text Colors
+    @"placeholderTextColor": @(0x4c3c3c43),
+    // Standard Content Background Colors
+    @"systemBackgroundColor": @(0xFFffffff),
+    @"secondarySystemBackgroundColor": @(0xFFf2f2f7),
+    @"tertiarySystemBackgroundColor": @(0xFFffffff),
+    // Grouped Content Background Colors
+    @"systemGroupedBackgroundColor": @(0xFFf2f2f7),
+    @"secondarySystemGroupedBackgroundColor": @(0xFFffffff),
+    @"tertiarySystemGroupedBackgroundColor": @(0xFFf2f2f7),
+    // Separator Colors
+    @"separatorColor": @(0x493c3c43),
+    @"opaqueSeparatorColor": @(0xFFc6c6c8),
+    // Link Color
+    @"linkColor": @(0xFF007aff),
+    // https://developer.apple.com/documentation/uikit/uicolor/standard_colors
+    // Adaptable Colors
+    @"systemBrownColor": @(0xFFa2845e),
+    @"systemIndigoColor": @(0xFF5856d6),
+    // Adaptable Gray Colors
+    @"systemGray2Color": @(0xFFaeaeb2),
+    @"systemGray3Color": @(0xFFc7c7cc),
+    @"systemGray4Color": @(0xFFd1d1d6),
+    @"systemGray5Color": @(0xFFe5e5ea),
+    @"systemGray6Color": @(0xFFf2f2f7),
+  };
+
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+  id savedTraitCollection = nil;
+  if (@available(iOS 13.0, *)) {
+    savedTraitCollection = [UITraitCollection currentTraitCollection];
+
+    [UITraitCollection setCurrentTraitCollection:[UITraitCollection traitCollectionWithUserInterfaceStyle:UIUserInterfaceStyleLight]];
+  }
+#endif
+
+  for (NSString *semanticColor in semanticColors) {
+    id json = RCTJSONParse([NSString stringWithFormat:@"{ \"semantic\": \"%@\" }", semanticColor], nil);
+    UIColor *value = [RCTConvert UIColor:json];
+    XCTAssertNotNil(value);
+
+    NSNumber *fallback = [semanticColors objectForKey:semanticColor];
+    NSUInteger rgbValue = [fallback unsignedIntegerValue];
+    NSUInteger alpha1 = ((rgbValue & 0xFF000000) >> 24);
+    NSUInteger red1 = ((rgbValue & 0x00FF0000) >> 16);
+    NSUInteger green1 = ((rgbValue & 0x0000FF00) >> 8);
+    NSUInteger blue1 = ((rgbValue & 0x000000FF) >> 0);
+
+    CGFloat rgba[4];
+    RCTGetRGBAColorComponents([value CGColor], rgba);
```

Comment:
also this is using `RCTGetRGBAColorComponents` which appears to check only for a couple `CGColorSpaceModel`s (`kCGColorSpaceModelMonochrome`, `kCGColorSpaceModelRGB`) so if we ever have any semantical color using another model it will fail.

Reply:
Only if they actually change the color model of one of the few semantic colors being used in the tests.